### ISSUE TITLE
Add possibility to assert value as they come through onChanged

### ISSUE THREAD

**Issue by vitorhugods** [jraska/livedata-testing#35]
I am trying to test mutable data that is stored in a LiveData.

But I couldn't find a proper way of asserting data fields as they come.
In order to really assert the value as it comes, I had to use the `map` method in a way it probably wasn't designed to. It feels really awkward using it, even though it works.

Example:

```kotlin
    @Test
    fun testMutableData() {
        val testSubject = subject.measurementEventData.test()
                .map { measurementEvent ->
                    // This is the part that feels awkward
                    measurementEvent.isHandled shouldBe false
                    measurementEvent
                }
        
        val measurement = Measurement(Sensor.TEMPERATURE, 0.55)
        
        subject.measurementEventData.value = Event(measurement)

        testSubject.value().isHandled = true

        testSubject
                .assertValue { measurementEvent ->
                    measurementEvent.isHandled
                }
    }
```

I propose to add to the TestObserver a list of asserters and test them in every `onChanged` alongside the `childObservers`.

**Comment by jraska** [comment]
Hello Vitor,

Thanks for the feedback and ideas. Truly the `map` had completely different intent. :) 

I have some thoughts about how to implement this and I think it might be useful feature. I will implement it once I have some time of PR is alway welcome :)

Josef

**Comment by jraska** [comment]
Hey again, I implemented possible solution [here](https://github.com/jraska/livedata-testing/commit/01d9902f16a4a86fc6af932e200c738510711b90#diff-54a9f5f6350826c0c1faa1dbb38f9a12R199), however I realised one important thing during implementing this.

All we need for your use case is something(e.g. assertion mentioned by you) which accepts value and is triggered each time new value is received by `LiveData`. After implementing hook into `TestObserver` I realised that is exactly what `Observer` is :D 

Therefore I think it might not be worth adding this into library, because test mentioned by you can be rewritten like:

```
  @Test
  fun testMutableData() {
    subject.measurementEventData.observeForever{ event -> event.isHandled shouldBe false}

    val measurement = Measurement("Temperature", 0.55)

    subject.measurementEventData.value = Event(measurement)
    subject.measurementEventData.value.isHandled = true

    subject.measurementEventData.test()
      .assertValue { measurementEvent ->
        measurementEvent.isHandled
      }
  }
```

Basically you might not need this library for assertions on single incoming values.

I'm still thinking if it is worth to have this hook implemented into library for convenience and more fluent testing, but I'm curious what do you think?
